src/dotnet/Fable.Compiler/Transforms/FableTransforms.fs

```diff
@@ -540,10 +540,18 @@ let optimizeExpr (com: ICompiler) e =
     List.fold (fun e f -> f com e) e optimizations
 
 let rec optimizeDeclaration (com: ICompiler) = function
+
+#if FABLE_COMPILER
+    // optimization for actions and values does not result in a valid expr
+    // TODO: enable optimization when it's fixed
+    | ActionDeclaration expr -> ActionDeclaration expr
+    | ValueDeclaration(value, info) -> ValueDeclaration(value, info)
+#else
```

Comment:
It'd be good to find why this is failing as ValueDeclaration includes most of the functions and optimizations do a lot of inlining. At a minimum we should try to apply the uncurrying optimizations because they're necessary to interact with fable-core files (in Fable 1 the uncurrying was done in Replacements module, but Fable 2 does it here).

Reply:
@alfonsogarciacaro For whatever reason, applying any optimization for action and value resuts in invalid expr.

Reply:
Is it happening in any case? At what point do you see it's an invalid expression? Do you have any example?

Anyway, I'll try to debug later :)